lib/common/common/src/counter/hardware_accumulator.rs

```diff
@@ -47,3 +101,46 @@ impl Default for HwMeasurementAcc {
         }
     }
 }
+
+/// A type collecting hardware measuerements in multi threaded scenarios where multiple instances of `HwMeasurementAcc` would be needed.
+/// Because `HwMeasurementAcc` purposely doesn't implement Clone, we need this type to allow having an owned type we can move
+/// between threads/async closures.
+///
+/// This collector automatically applies all measurements to the `HwMeasurementAcc` that was used when creating this collector on drop.
+/// To ensure all values get applied, before reading from a `HwMeasurementAcc`, this type must be dropped first.
+pub struct HwMeasurementCollector {
+    /// Shared reference of CPU usage with the given Parent.
+    cpu_counter: Arc<AtomicUsize>,
+
+    /// A new temporary measurement accumulator, which values get applied to the parents
+    /// ones on drop.
+    collector: HwMeasurementAcc,
+}
+
+impl HwMeasurementCollector {
+    fn new(cpu_counter: Arc<AtomicUsize>) -> Self {
+        let collector = HwMeasurementAcc::new();
+        Self {
+            cpu_counter,
+            collector,
+        }
+    }
+}
+
+impl Deref for HwMeasurementCollector {
+    type Target = HwMeasurementAcc;
```

Comment:
same as implicit conversion, it is quite impossible to find where exactly this deref is used.